### DISCUSSION TITLE
Allow Auto Merge to wait on non-required checks

### DISCUSSION THREAD

**Discussion by cdmcmahon** [community/community#27349] [Pull Requests]
## Background

We have a number of checks that run for all PRs in our organization. While some are required, others are not (linting, certain classes of tests), in order to enable quick merging in certain situations. However, we have a best practice of letting all checks run in order to ensure the main branch stays healthy. We'd like to avoid requiring these checks for the aforementioned reasons, but it seems that this blocks the utility of Auto Merge for us as it will immediately merge when all required checks pass, even if non-required checks are still running.

## Feature Request

An option when Auto Merge is enabled to merge after required checks or merge. For our use case, a repository wide setting would be sufficient, not on an individual PR.

_(Note: I looked pretty thoroughly for this issue and didn't find anything, even though I had assumed it might already be requested. Apologies if I missed it.)_

**Comment by corey-tock** [comment]
Of course after posting I find [#7114 ](https://github.com/community/community/discussions/7114) but that specifically mentions Github Actions. I believe we include non-Github Actions checks, as well, so this is slightly more broad.

**Comment by sebwills** [comment]
I agree this would be useful. I don't want to make all my checks required as that would block humans merging with failed checks if necessary. But if I don't, auto merge merges prematurely.

Perhaps the most flexible solution would be to allow configuration of a separate list of required status checks for auto merge? So some checks could be required for auto merge but not for humans.

**Comment by adamu** [comment]
I maintain an app, [Mergery](https://github.com/apps/mergery), which behaves this way.

**Comment by Magnuti** [comment]
Still not fixed after 4 years?
